helix-core/src/main/java/org/apache/helix/controller/stages/CustomizedStateOutput.java

```diff
@@ -0,0 +1,126 @@
+package org.apache.helix.controller.stages;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.apache.helix.model.CustomizedState;
+import org.apache.helix.model.Partition;
+
+public class CustomizedStateOutput {
+  // stateType -> (resourceName -> (Partition -> (instanceName -> customizedState)))
+  private final Map<String, Map<String, Map<Partition, Map<String, String>>>> _customizedStateMap;
+
+  private final Map<String, CustomizedState> _customizedStateMetaMap;
```

Comment:
How this is used? I would suggest adding this only when we have the usage in the PR.